src/Control/Carrier/NonDet/Church.hs

```diff
@@ -109,33 +108,5 @@ instance MonadTrans NonDetC where
 instance (Algebra sig m, Effect sig) => Algebra (NonDet :+: sig) (NonDetC m) where
   eff (L (L Empty))      = empty
   eff (L (R (Choose k))) = k True <|> k False
-  eff (R other)          = NonDetC $ \ fork leaf nil -> eff (handle (Leaf ()) (fmap join . traverse runNonDetA) other) >>= fold fork leaf nil
+  eff (R other)          = NonDetC $ \ fork leaf nil -> eff (handle (pure ()) (runNonDet (liftA2 (<|>)) runNonDetA (pure empty)) other) >>= runNonDet fork leaf nil
   {-# INLINE eff #-}
-
-
-data BinaryTree a = Nil | Leaf a | Fork (BinaryTree a) (BinaryTree a)
```

Comment:
Now that this is gone, can you update the `Changelog`’s entry for `NonDetC` and remove the mention of a second binary tree? Such a mention is also present in `Control.Carrier.Choose.Church` and `Control.Carrier.NonDet.Church`.

Reply:
I don’t follow. This is the only mention I’m seeing in the changelog:

> - Redefines `NonDetC` as a Church-encoded binary tree instead of a Church-encoded list ([#197](https://github.com/fused-effects/fused-effects/pull/197)).

That, and the comments in the carrier modules, are still accurate—it’s a Church-encoded binary tree. Contrast the church-encoded binary tree handler:

```haskell
runNonDet
  :: (m b -> m b -> m b) -- fork
  -> (a -> m b)          -- leaf
  -> m b                 -- nil
  -> NonDetC m a
  -> m b
```

with the church-encoded list handler:

```haskell
runNonDet
  :: (a -> m b -> m b) -- cons
  -> m b               -- nil
  -> NonDetC m a
  -> m b
```

Reply:
I don’t follow. This is the only mention I’m seeing in the changelog:

> - Redefines `NonDetC` as a Church-encoded binary tree instead of a Church-encoded list ([#197](https://github.com/fused-effects/fused-effects/pull/197)).

That, and the comments in the carrier modules, are still accurate—it’s a Church-encoded binary tree. Contrast the church-encoded binary tree handler:

```haskell
runNonDet
  :: (m b -> m b -> m b) -- fork
  -> (a -> m b)          -- leaf
  -> m b                 -- nil
  -> NonDetC m a
  -> m b
```

with the church-encoded list handler:

```haskell
runNonDet
  :: (a -> m b -> m b) -- cons
  -> m b               -- nil
  -> NonDetC m a
  -> m b
```